unicon_backend/models/problem.py

```diff
@@ -83,6 +87,22 @@ def _convert_task_to_orm(id: int, type: TaskType, autograde: bool, **other_field
 
         return _convert_task_to_orm(**task.model_dump(serialize_as_any=True))
 
+    def to_task(self) -> "Task":
+        task_classes: dict[TaskType, type[Task]] = {
+            TaskType.MULTIPLE_CHOICE: MultipleChoiceTask,
+            TaskType.MULTIPLE_RESPONSE: MultipleResponseTask,
+            TaskType.PROGRAMMING: ProgrammingTask,
+            TaskType.SHORT_ANSWER: ShortAnswerTask,
+        }
```

Comment:
we can probably shift this to be under `unicon_backend/evaluator/tasks/__init__.py` since this is going to be a static dict

_could use `typing.Final` or `collections.abc.Mapping` to enforce immutability